api/controllers/DashboardController.js

```diff
@@ -87,6 +53,50 @@ var dashboardController = {
         });
     },
 
+    _calculateSortParams(req, res, userData, totalApplications) {
+        const pageSize = 20;
+        const currentPage = req.query.page || 1;
+        const offset = pageSize * (currentPage - 1);
+        const sortOrder = Number(req.query.sortOrder) || -1;
+        const isNumberPositive = Math.sign(sortOrder) === 1;
+        const direction = isNumberPositive ? 'asc' : 'desc';
+        const searchCriteria =
+            req.allParams().dashboardFilter || req.query.searchText || '';
+        //If user has specifically selected to filter by date (sortOrder eq 1 or -1)
+        //then we don't want to secondary sort by date again! But if a user filters by
+        //reference number for example, then secondary sort on the date as well.
+        const secondarySortOrder =
+            (sortOrder === 1 || sortOrder === -1) ? null : '1';
+        const secondaryDirection =
+            (sortOrder === 1 || sortOrder === -1) ? null : 'desc';
```

Comment:
it seems like sortOrder is already casted with `|| -1`, so wont these ternaries always be true?

Reply:
It's super weird but you're partly right. When the page first loads the ternaries are true because `req.query.sortOrder` is undefined (line 60). But `req.query.sortOrder` always passes a string of a number, and if that number is not 1 (or -1), then they are false.

Reply:
How about something like this 
```suggestion
       let secondarySortOrder, secondaryDirection;
       
       if (sortOrder === 1 || sortOrder === -1) {
       ...
       }
       //or even
       const hasSortOrder = sortOrder === 1 || sortOrder === -1
       if(!hasSortOrder) {..} 
```
your comment already explains that but I usually skip over them (as you can tell from my confusion). 😄